### PR TITLE
Tslint fix

### PR DESCRIPTION
**Pull request by beygi**
add ```tslintFix``` option to getTypescriptCode() and getCustomCode() to fix generated code using tslint fixer

### CODE REVIEW

File: lib/codegen.js [hunk 1]
```diff
@@ -251,6 +252,18 @@ function enhanceCode(source, opts, type){
     } else {
         return source;
     }
+
+    if (opts.tslintFix === undefined || opts.tslintFix === true) {
```
Comment (mtennoe) [inline lib/codegen.js]: `opts.tslintFix === undefined` should probably be considered as not wanting to run tslintfix

File: templates/class.mustache [hunk 2]
```diff
@@ -2,10 +2,10 @@
 /// <reference path="{{&.}}" />
 {{/imports}}
 
-import request, { SuperAgentStatic, SuperAgentRequest, Response } from "superagent";
+import request, { SuperAgentStatic, SuperAgentRequest, Response, } from "superagent";
```
Comment (mtennoe) [inline templates/class.mustache]: Why the changes to the two template files? Those changes are not related to tslintfix being run

File: templates/class.mustache [hunk 3]
```diff
@@ -2,10 +2,10 @@
...
 
-export type RequestHeaders = {
-    [header: string]: string;
+export interface RequestHeaders  {
+    header: string[];
```
Comment (mtennoe) [inline templates/class.mustache]: This turns the header type from a dictionary to a string array

File: templates/class.mustache [hunk 4]
```diff
@@ -17,7 +17,6 @@ export type CallbackHandler = (err: any, res?: request.Response) => void;
 
 {{#definitions}}
 export type {{&name}} = {{#tsType}}{{> type}}{{/tsType}};
-
```
Comment (mtennoe) [inline templates/class.mustache]: This newline is there to make the output slightly nicer

File: lib/codegen.js (hunk 1 above)
Comment (mtennoe) [inline lib/codegen.js]: @Markionium - This needs to be considered for the typescript rewrite

**Comment by mtennoe** [comment]
@beygi  - Thanks for contributing!

**Comment by Markionium** [comment]
@beygi Is there a reason why you closed this? Do you still wish to get this merged?

**Comment by beygi** [comment]
@Markionium it seems tslint lib is unable to fix source code on the fly. it just working on files.
i will implement this with temp files and add another option to pass tslint rules to options then i create another pull request for all of those changes.

**Comment by Markionium** [comment]
Just to make sure you're aware. My massive PR will go in not too long from
now. You might want to base your changes on that one to prevent yourself
from having to resolve a massive merge conflict.

It's the only currently open PR that rewrites the whole project to
typescript.

**Comment by Markionium** [comment]
@beygi seems like @mtennoe just merged it. :)

So you can forget what I said above and just work on master :)

**Comment by mtennoe** [comment]
Yup, just merged and published, so it is ready to be working on top of :)

**Comment by beygi** [comment]
cool, moving the project to typescript was a great idea
